src/core/shape_inference/include/broadcast_shape_inference.hpp

```diff
@@ -49,118 +49,112 @@ void validate_target_shape_none(const ov::Node* op,
     using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
-    if (arg0_shape.rank().is_dynamic() || target_shape.rank().is_dynamic()) {
-        result_shape = PartialShape::dynamic(target_shape.rank());
+    if (arg0_shape.rank().is_dynamic() || target_input_shape.rank().is_dynamic()) {
+        result_shape = PartialShape::dynamic(target_input_shape.rank());
         return;
     }
-    result_shape = target_shape;
+    result_shape = target_input_shape;
     auto& start_axis = broadcast_spec.m_axis;
 
     NODE_VALIDATION_CHECK(op, start_axis >= 0, "Broadcast start_axis must be greater than 0");
 
-    for (size_t i = start_axis; i < target_shape.size(); i++) {
+    for (size_t i = start_axis; i < target_input_shape.size(); i++) {
         const auto& arg_dim = arg0_shape[i - start_axis];
         if (arg_dim == 1) {
-            result_shape[i] = target_shape[i];
-        } else if (target_shape[i] == 1) {
+            result_shape[i] = target_input_shape[i];
+        } else if (target_input_shape[i] == 1) {
             result_shape[i] = arg_dim;
         } else {
             NODE_VALIDATION_CHECK(op,
-                                  DimType::merge(result_shape[i], arg_dim, target_shape[i]),
+                                  DimType::merge(result_shape[i], arg_dim, target_input_shape[i]),
                                   "Broadcast incorrect target shape. Expecting either 1 or ",
                                   arg_dim,
                                   " . Got ",
-                                  target_shape[i]);
+                                  target_input_shape[i]);
         }
     }
 }
 
 template <typename T>
-void set_result_shape_bidirectional(const ov::Node* op, const T& arg_shape, T& target_shape, T& result_shape) {
+void set_result_shape_bidirectional(const ov::Node* op, const T& arg_shape, T& target_input_shape, T& result_shape) {
     using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
-    if (arg_shape.rank().is_dynamic() || target_shape.rank().is_dynamic()) {
+    if (arg_shape.rank().is_dynamic() || target_input_shape.rank().is_dynamic()) {
         result_shape = PartialShape::dynamic();
         return;
     }
     auto arg_shape_vec = arg_shape;
 
     // Add left padding to shorter target or argument shape
-    const auto target_padded_rank = std::max(arg_shape_vec.size(), target_shape.size());
-    while (arg_shape_vec.size() < target_padded_rank) {
-        arg_shape_vec.insert(arg_shape_vec.begin(), 1);
-    }
-    while (target_shape.size() < target_padded_rank) {
-        target_shape.insert(target_shape.begin(), 1);
-    }
+    const auto target_padded_rank = std::max(arg_shape_vec.size(), target_input_shape.size());
+
+    if (arg_shape_vec.size() < target_padded_rank)
+        arg_shape_vec.insert(arg_shape_vec.begin(), target_padded_rank - arg_shape_vec.size(), 1);
+    else if (target_input_shape.size() < target_padded_rank)
```

Comment:
Even before, it was not possible that both conditions are `true`, because the `target_padded_rank` is either `target_shape.size()` or `arg_shape_vec.size()`. 
It can't be smaller and equal at the same time.

Reply:
But then redundant `insert` with `0` as count will be called if `arg_shape_vec.size() == target_input_shape.size()`. 
Is it okay and "costless"?

Reply:
For count `0` nothing happens but count somehow is checked ( it think cost of this check is minor). 
Currently you check size multiple times (line with `std::max` is not required at all) and given example give smaller code.

You can current logic but without `target_padded_rank` and there will be no insertion call at all. 
Count is validated explicitly instead using `insert` internal stuff. Is minor think, is up to you what you select.

Reply:
Okay, removed `target_padded_rank` and applied the latest suggestion.